devito/types/petsc.py

```diff
@@ -66,6 +66,15 @@ class PETScArray(ArrayBasic):
 
     _data_alignment = False
 
+    __rkwargs__ = (ArrayBasic.__rkwargs__ +
+                   ('liveness',))
+
+    def __init_finalize__(self, *args, **kwargs):
+        super().__init_finalize__(*args, **kwargs)
+
+        self._liveness = kwargs.get('liveness', 'lazy')
```

Comment:
This goes above super with pop to avoid unknown kwargs

Reply:
Thanks. Is it GTG now?